src/v/reflection/async_adl.h

```diff
@@ -33,6 +33,48 @@ struct async_adl {
     }
 };
 
+namespace detail {
+
+template<template<class...> class GenericMap, class K, class V, class... Args>
+struct async_adl_map {
+    using Map = GenericMap<K, V, Args...>;
+
+    ss::future<> to(iobuf& out, Map&& map) {
+        reflection::serialize(out, static_cast<int32_t>(map.size()));
+        return ss::do_with(std::move(map), [&out](Map& map) {
+            return ss::do_for_each(map, [&out](typename Map::value_type& p) {
+                /// We must copy the key since map keys are all const
+                auto copy = p.first;
```

Comment:
use `extract()` to get a mutable reference to the key?

Reply:
Using extract while iterating would invalidate iterators like map::erase().